Java/medical clinic/med 11

HubertKarw opened the pull request.

HubertKarw committed:
java/medical-clinic/task1/patient control
added methods for displaying and editing patients

HubertKarw committed:
java/medical-clinic/med11/validation
patient validator class and implementation

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Repository/PatientRepository.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    public void removePatient(String email) {
+        Patient patient = patients.stream()
```
mogles uzyc metody z repozytorium -> findByEmail()

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Repository/PatientRepository.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    public Patient modifyPatient(String email, Patient newPatient) {
+        Patient patient = patients.stream()
```
mogles uzyc metody z repozytorium -> findByEmail()

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,32 @@
+    }
+    public static void validateAddingPatient(List<Patient> patients, String email){
+        if(!patients.isEmpty()){
+            if(patients.stream().map(Patient::getEmail).anyMatch(patientEmail -> patientEmail.equals(email))){
```
tak nie formatujemy streamów

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,32 @@
+            throw new IllegalArgumentException("Patient cannot change idCardNo");
+        }
+    }
+    public static void validateAddingPatient(List<Patient> patients, String email){
```
ctrl alt l

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,32 @@
+        if (!patient.getIdCardNo().equals(anotherPatient.getIdCardNo())) {
+            throw new IllegalArgumentException("Patient cannot change idCardNo");
+        }
+    }
```
brak odstepu miedzy metodami

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,32 @@
+            throw new IllegalArgumentException("Patient cannot change idCardNo");
+        }
+    }
+    public static void validateAddingPatient(List<Patient> patients, String email){
```
uzyjmy repozytorium i metody findByEmail

konradgab replied:
oprocz tego zastanow sie nad przypadkiem kiedy pacjent nie zmienia maila czyli w body idzie taki mail jaki ma aktualnie

HubertKarw committed:
java/medical-clinic/med04/ password update and validation  changes
new validation method to udpate email

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,48 @@
+
+    public static void validateEmailUpdate(Patient patient, Patient anotherPatient, PatientRepository repository) {
+        if (!patient.getEmail().equals(anotherPatient.getEmail())) {
+            List<Patient> otherPatients = repository.getPatients();
```
Optional optionalPatient = patientRepository.findByEmail(patient.getEmail());
if (optionalPatient .isPresent() && !optionalPatient.get().getEmail().equals(email)) {
throw new IllegalArgumentException("Patient with given email already exists");
}

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,48 @@
+        List<Patient> patients = repository.getPatients();
+
+        if (!patients.isEmpty()) {
+            if (patients.stream()
```
czemu by nie uzyc findByEmail zeby repoztyroium nam zwrocilo czy taki pacjent jest czy go nie ma

konradgab commented:
approved

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -0,0 +1,49 @@
+package com.HubertKarw.medical_clinic.Controller;
+
+import com.HubertKarw.medical_clinic.Model.Password;
+import com.HubertKarw.medical_clinic.Model.Patient;
```
xx

HubertKarw committed:
java/medical-clinic/med08/ Exception classes and handler
added PatientCreationException and PatientNotFoundException classes and MedicalClinicExceptionHandler handler class

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -0,0 +1,49 @@
+import com.HubertKarw.medical_clinic.Model.Patient;
+import com.HubertKarw.medical_clinic.Service.PatientService;
+import lombok.RequiredArgsConstructor;
+import org.apache.catalina.User;
```
co to jest

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Exception/MedicalClinicExceptionHandler.java:
```diff
@@ -0,0 +1,22 @@
+
+@ControllerAdvice
+public class MedicalClinicExceptionHandler {
+    @ExceptionHandler(PatientNotFoundException.class)
```
nie robimy takich skrotow jak juz to ex -> exception

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Exception/PatientCreationException.java:
```diff
@@ -0,0 +1,7 @@
+package com.HubertKarw.medical_clinic.Exception;
+
+public class PatientCreationException extends RuntimeException {
+    public PatientCreationException(String s){
```
String message

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Exception/MedicalClinicExceptionHandler.java:
```diff
@@ -0,0 +1,22 @@
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class MedicalClinicExceptionHandler {
```
Moglibysmy stworzyć MedicalClinicException ktory dziedziczy po runtime exception i kazdy z tych wyjątków opisanych tutaj dziedziczylby nie po RuntimeException tylko po MedicalClinicException i w MedicalClinicException dodac pole HTTPStatus dzieki czemu kazdy wyjatek w naszej aplikacji bedzie mozna powiedziec takim Webowym wyjątkiem, ktory posiada message oraz status

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Exception/MedicalClinicExceptionHandler.java:
```diff
@@ -0,0 +1,22 @@
+@ControllerAdvice
+public class MedicalClinicExceptionHandler {
+    @ExceptionHandler(PatientNotFoundException.class)
+    ResponseEntity<String> handlePatientNotFound(PatientNotFoundException pnfe){
```
ResponseEntity<String> -> ResponseEntity<ApiError>

HubertKarw committed:
java/medical-clinic/med08/ DTO classes
added dto classes and mapper

HubertKarw committed:
java/medical-clinic/med08/ DTO classes
added dto classes and mapper

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -0,0 +1,54 @@
+        return PatientMapper.toDTO(patientService.addPatient(PatientMapper.toPatient(command)));
+    }
+
+    @DeleteMapping("/{email}")
```
@ResponseStatus(NO_CONTENT)

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -0,0 +1,54 @@
+        return PatientMapper.toDTO(patientService.getPatient(email));
+    }
+
+    @PostMapping
```
@ResponseStauts(CREATED)

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Validation/PatientValidator.java:
```diff
@@ -0,0 +1,55 @@
+        List<Patient> patients = repository.getPatients();
+
+        if (!patients.isEmpty()) {
+//            if (patients.stream()
```
komentarz do usuniecia nie zostawiamy zakomentowanego kodu

HubertKarw committed:
java/medical-clinic/med08/ user classes (repository, service, controller, DTO and command), review changes and structMapper
created classes for users and mappers for users and patients

HubertKarw committed:
java/medical-clinic/med08/ medical clinic exception

HubertKarw committed:
java/medical-clinic/med08/ medical clinic exception

HubertKarw committed:
java/medical-clinic/med08/ medical clinic exception

HubertKarw committed:
java/medical-clinic/med08/ Swagger implementation

HubertKarw committed:
java/medical-clinic/med08/ Swagger implementation

HubertKarw committed:
java/medical-clinic/med08/ Swagger implementation

HubertKarw committed:
java/medical-clinic/med08/ jpaRepo, service and validation implementation

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
 
+    @Operation(summary = "Get all patients")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found patients", content = {@Content(mediaType = "application/json",schema = @Schema(implementation = PatientDTO.class))})})
```
Return all patients

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
+            description = "Patient to create", required = true,
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = CreatePatientCommand.class),
+                    examples = @ExampleObject(value = "{ \"Email\": \"email\", \"User\": \"User\", \"idCardNO\": \"1234\"" +
```
idCardNo

konradgab replied:
klucze do poprawy

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
-
+    @Operation(summary = "Remove patient")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "removed patient", content = {@Content}),
```
Removed

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
+        return mapper.mapToDTO(patientService.addPatient(mapper.mapToPatient(command)));
     }
-
+    @Operation(summary = "Remove patient")
```
Remove patient by email

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
+    @Operation(summary = "Get patient by email")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found patient", content = {@Content(mediaType = "application/json",schema = @Schema(implementation = PatientDTO.class))}),
+            @ApiResponse(responseCode = "404", description = "Patient not found", content = @Content)
```
ErrorMessage jako schema + mediatype app json

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
-
+    @Operation(summary = "Get patient by email")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found patient", content = {@Content(mediaType = "application/json",schema = @Schema(implementation = PatientDTO.class))}),
```
ctrl alt l

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
-
+    @Operation(summary = "create patient")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Created patient", content = {@Content(mediaType = "application/json",schema = @Schema(implementation = PatientDTO.class))}),
```
ctrl alt l

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
+    @Operation(summary = "create patient")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Created patient", content = {@Content(mediaType = "application/json",schema = @Schema(implementation = PatientDTO.class))}),
+            @ApiResponse(responseCode = "400", description = "Patient is not valid", content = @Content)
```
ErrorMessage jako schema + mediatype app json

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -4,51 +4,110 @@
         patientService.removePatient(email);
     }
-
+    @Operation(summary = "edit patient")
```
Edit patient

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Repository/PatientJpaRepository.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+@Repository
```
brak odstepu miedzy adnotacja a importem

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Repository/PatientJpaRepository.java:
```diff
@@ -0,0 +1,12 @@
+import java.util.Optional;
+@Repository
+public interface PatientJpaRepository extends JpaRepository<Patient,Long> {
+    public Optional<Patient> findByEmail(String email);
```
public jest zbedny poniewaz jest to interfejs

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Service/PatientJpaService.java:
```diff
@@ -0,0 +1,44 @@
+        return repository.findByEmail(email)
+                .orElseThrow(()->new PatientNotFoundException("Patient with this email does not exist"));
+    }
+    public Patient addPatient(Patient patient){
```
brak odstepu miedzy metodami

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Service/PatientJpaService.java:
```diff
@@ -0,0 +1,44 @@
+    }
+    public Patient getPatient(String email){
+        return repository.findByEmail(email)
+                .orElseThrow(()->new PatientNotFoundException("Patient with this email does not exist"));
```
ctrl alt l

HubertKarw committed:
java/medical-clinic/med08/formatting

HubertKarw committed:
java/medical-clinic/med08/USER TABLE IMPLEMENTATION

HubertKarw committed:
java/medical-clinic/med08/doctor and institution implementation

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/DoctorController.java:
```diff
@@ -0,0 +1,71 @@
+    @Operation(summary = "Get all doctors")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found patients", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = Doctor.class))})})
+    @GetMapping()
```
@GetMapping

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/DoctorController.java:
```diff
@@ -0,0 +1,71 @@
+        return mapper.mapToDTO(service.modifyDoctor(email, mapper.mapToDoctor(command)));
+    }
+
+    @PatchMapping("/{email}/{name}")
```
PATCH /doctors/kg@gmail.com/institutions/konrad

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/DoctorController.java:
```diff
@@ -0,0 +1,71 @@
+    @GetMapping()
+    public List<DoctorDTO> getDoctors() {
+        return service.getDoctors().stream()
+                .peek(System.out::println)
```
do wyrzucenia

HubertKarw committed:
java/medical-clinic/med08/doctor and institution implementation

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/DoctorController.java:
```diff
@@ -32,8 +35,10 @@ public class DoctorController {
     @GetMapping()
-    public List<DoctorDTO> getDoctors() {
-        return service.getDoctors().stream()
+    public List<DoctorDTO> getDoctors(@RequestParam(name = "page", defaultValue = "0") int page) {
```
Brakuje parametru size

konradgab replied:
Przyjmijmy Pageable zamiast page oraz size

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/InstitutionController.java:
```diff
@@ -23,8 +25,10 @@ public class InstitutionController {
     @GetMapping
-    public List<InstitutionDTO> getInsitutions() {
-        return service.getInstitutions().stream()
+    public List<InstitutionDTO> getInsitutions(@RequestParam(name = "page", defaultValue = "0") int page) {
```
Przyjmijmy Pageable

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/PatientController.java:
```diff
@@ -33,8 +35,10 @@ public class PatientController {
     @GetMapping
-    public List<PatientDTO> getPatients() {
-        return patientService.getPatients().stream()
+    public List<PatientDTO> getPatients(@RequestParam(name = "page", defaultValue = "0") int page) {
```
Przyjmijmy Pageable

konradgab commented on src/main/java/com/HubertKarw/medical_clinic/Controller/UserController.java:
```diff
@@ -34,8 +36,10 @@ public class UserController {
     @GetMapping
-    public List<UserDTO> getUsers() {
-        return userService.getUsers().stream().map(mapper::toDto).collect(Collectors.toList());
+    public List<UserDTO> getUsers(@RequestParam(name = "page", defaultValue = "0") int page) {
```
Przyjmijmy Pageable

HubertKarw committed:
java/medical-clinic/med11/review changes

HubertKarw committed:
java/medical-clinic/med11/review changes
